org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/ScopeCompileTest.scala

```diff
@@ -61,23 +62,13 @@ object ScopeCompileTest {
   }
 
   @BeforeClass def setup(): Unit = {
-    SDTTestUtils.enableAutoBuild(false)
-    var done = false
-    List(projectAName, projectBName).foreach { name =>
-      EclipseUtils.workspaceRunnableIn(workspace) { monitor =>
-        val wspaceLoc = workspace.getRoot.getLocation
-        val src = new File(sourceWorkspaceLoc(bundleName).toFile().getAbsolutePath + File.separatorChar + name)
-        val dst = new File(wspaceLoc.toFile().getAbsolutePath + File.separatorChar + name)
-        FileUtils.copyDirectory(src, dst)
-        done = true
-      }
+    initializeProjects(bundleName, Seq(projectAName, projectBName)) {
+      projectA = createProjectInWorkspace(projectAName, withSrcOutputStructure)
+      projectB = createProjectInWorkspace(projectBName, withSrcOutputStructure)
+      addToClasspath(projectB, JavaCore.newProjectEntry(projectA.underlying.getFullPath, false))
+      projectA.underlying.refreshLocal(IResource.DEPTH_INFINITE, null)
+      projectB.underlying.refreshLocal(IResource.DEPTH_INFINITE, null)
```

Comment:
these parts are similar in fact but I treat them as a specific test description. In other test I can have let say 3 projects and for example no dependencies between them. I think it would be refactored maybe to some default configuration like 2 scala projects where B depends on A. Not a case here. I'm checking is 'refreshLocal' is really needed (I added it a little bit defensively)